gemac/management.py

```diff
@@ -0,0 +1,165 @@
+    status.(TriStateBuffer Connected)
+mdioOut - Output data signal from PHY for its configuration and
+    status.(TriStateBuffer Connected)
+mdioTri - TriState Control for Signals - Low Indicating mdioOut to be
+    asserted to the MDIO bus.
+
+"""
+
+
+@block
+def management(host_interface, mdio_interface):
+
+    configregisters = [Signal(intbv(0)[32:]) for _ in range(10)]
+    addresstable = [Signal(intbv(0)[48:]) for _ in range(4)]
+    addrtableread = Signal(bool(0))
+    addrtablelocation = Signal(intbv(0)[2:])
+    mdiodata = mdioData()
+
+    def getregisternumber(addr):
+        if addr >= 0x200 and addr <= 0x23F:
+            return rx0
+        elif addr >= 0x240 and addr <= 0x27F:
+            return rx1
+        elif addr >= 0x280 and addr <= 0x2BF:
+            return tx
+        elif addr >= 0x2C0 and addr <= 0x2FF:
+            return flow
+        elif addr >= 0x340 and addr <= 0x37F:
+            return managementreg
+        elif addr >= 0x380 and addr <= 0x383:
+            return ucast0
+        elif addr >= 0x384 and addr <= 0x387:
+            return ucast1
+        elif addr >= 0x388 and addr <= 0x38B:
+            return addrtable0
+        elif addr >= 0x38C and addr <= 0x38F:
+            return addrtable1
+        elif addr >= 0x390 and addr <= 0x393:
+            return addrfiltermode
+        else:
+            return reserved
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def readData():
+        if (not host_interface.miimsel) and host_interface.regaddress[9]:
+            regindex = getregisternumber(host_interface.regaddress)
+            if regindex != reserved and host_interface.opcode[1]:  # Read Config
+                host_interface.rddata.next = configregisters[regindex]
+            if regindex == addrtable1 and not host_interface.opcode[1] and \
+                    host_interface.wrdata[23]:  # Address Table Read 0
+                loc = host_interface.wrdata[18:16]
+                addrtablelocation.next = loc
+                host_interface.rddata.next = addresstable[loc][32:0]
+                addrtableread.next = True
+        if not host_interface.miimrdy and mdiodata.done:  # MDIO Read
+            host_interface.rddata.next = mdiodata.rddata[16:]
+        if addrtableread:  # Address Table Read 1
+            addrtableread.next = False
+            host_interface.rddata.next = addresstable[addrtablelocation][48:32]
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def writeConfig():
+        if (not host_interface.miimsel) and host_interface.regaddress[9] and \
+                (not host_interface.opcode[1]):
+            regindex = getregisternumber(host_interface.regaddress)
+            if regindex != reserved:
+                configregisters[regindex].next = host_interface.wrdata
+            if regindex == addrtable1 and not host_interface.wrdata[23]:
+                addresstable[host_interface.wrdata[18:16]].next = \
```

Comment:
Is the `host_interface.wrdata` set to a specific width (only supports 32 bit width)?  Where is the register information, definition, summary.  Is the register map documented somewhere?